test/booker_test.rb

```diff
@@ -0,0 +1,165 @@
+require_relative "test_helper"
+
+describe "Booker class" do 
+   
+  describe "Booker initialization" do
+    
+    it "throws argument erros for invalid inventory input" do
+      expect{Hotel::Booker.new(inventory: '20')}.must_raise ArgumentError
+      expect{Hotel::Booker.new(inventory: 0)}.must_raise ArgumentError
+    end
+    
+    before do
+      @new_booker = Hotel::Booker.new(inventory: 20)
+    end
+    
+    it "is an instance of Booker" do 
+      expect(@new_booker).must_be_kind_of Hotel::Booker
+    end
+
+    it "has an empty array for reservations" do 
+      expect(@new_booker.reservations).must_be_kind_of Array
+      expect(@new_booker.reservations.length).must_equal 0
+    end
+
+    it "has an array of Room instances matching with the inventory number" do
+      expect(@new_booker.rooms).must_be_kind_of Array
+      expect(@new_booker.rooms.first).must_be_kind_of Hotel::Room
+      expect(@new_booker.rooms.last).must_be_kind_of Hotel::Room
+      expect(@new_booker.rooms.length).must_equal 20
+    end
+
+  end
+  
+  describe "all_rooms method" do 
+    it "must return an array of Room instances matching with the inventory number" do
+      expect(Hotel::Booker.new(inventory: 33).all_rooms).must_be_kind_of Array
+      expect(Hotel::Booker.new(inventory: 33).all_rooms.first).must_be_kind_of Hotel::Room
+      expect(Hotel::Booker.new(inventory: 33).all_rooms.last).must_be_kind_of Hotel:: Room
+      expect(Hotel::Booker.new(inventory: 33).all_rooms.length).must_equal 33
+    end
+
+  end
+
+  describe "available_rooms method" do
+    before do
+      @new_booker = Hotel::Booker.new(inventory: 2)
+    end
+    
+    it "throws ArgumentError for invalid check_in and check_out dates" do
+      expect{ @new_booker.make_reservation(check_in: "2019-09-01", check_out: "2019-09-01") }.must_raise ArgumentError
+      expect{ @new_booker.make_reservation(check_in: "2019-09-02", check_out: "2019-09-01") }.must_raise ArgumentError
+    end
+    
+    it "throws ArgumentError when there is no available rooms during the given date range" do 
+      rooms = @new_booker.rooms
+      rooms[0].bookings << Hotel::Reservation.new(check_in: "2019-09-01", check_out: "2019-09-04", room: 1)
```

Comment:
Consider storing the dates in variables so it is clear what you are testing.
```suggestion
      check_in_date = "2019-09-01"
      check_out_date = "2019-09-04"
```